### PR TITLE
Publish ClusterOperator status.

### PR DESCRIPTION
**Pull request by jmencak**
This PR adds support for publishing ClusterOperator status
https://github.com/openshift/api/blob/master/config/v1/types_cluster_operator.go#L33

### CODE REVIEW

File: pkg/util/clusteroperator/status.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		newConditions = append(newConditions, *condition)
+	}
+
+	return newConditions
+}
+
+// ConditionsEqual returns true if and only if the provided slices of conditions
+// (ignoring LastTransitionTime) are equal.
+func ConditionsEqual(oldConditions, newConditions []configv1.ClusterOperatorStatusCondition) bool {
+	if len(newConditions) != len(oldConditions) {
```
Comment (sjug) [inline pkg/util/clusteroperator/status.go]: Would this ever be true?

File: pkg/controller/tuned/status.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		} else {
+			availableCondition.Status = configv1.ConditionFalse
+			availableCondition.Message = fmt.Sprintf("DaemonSet %q has %d unavailable pod(s).", daemonset.Name, daemonset.Status.NumberUnavailable)
+			progressingCondition.Status = configv1.ConditionTrue
+
+		}
+		requeue = daemonset.Status.NumberReady != daemonset.Status.DesiredNumberScheduled
+	}
+
+	conditions = clusteroperator.SetStatusCondition(conditions, availableCondition)
```
[the remark this answers is not in the document]
Comment (jmencak) [reply pkg/controller/tuned/status.go]: I'm afraid, looking at the API https://github.com/openshift/api/blob/master/config/v1/types_cluster_operator.go this is not clear to me.  The best I can do is to follow what most operators do, and they seem to be setting all 3 conditions.

File: pkg/util/clusteroperator/status.go (hunk 1 above)
Comment (jmencak) [reply pkg/util/clusteroperator/status.go]: Not sure what you're asking or what the problem is here.  Function or the `if` statement?  In any case, the function can return `true/false` and the `if` statement can evaluate to `true` or `false`.

File: pkg/config/config.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	return "cluster-node-tuning-operator" // return default operator name unless defined in the environment
+}
+
+// NodeTunedImage returns the operator's operand/tuned image path.
+func NodeTunedImage() string {
+	imageTuned := os.Getenv("CLUSTER_NODE_TUNED_IMAGE")
+
+	return imageTuned
```
[the remark this answers is not in the document]
Comment (jmencak) [reply pkg/config/config.go]: Agreed, will change.

File: pkg/controller/tuned/status.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+)
+
+// syncOperatorStatus computes the operator's current status and therefrom
+// creates or updates the ClusterOperator resource for the operator.
+func (r *ReconcileTuned) syncOperatorStatus() (bool, error) {
+	var requeue bool
+
+	log.Printf("syncOperatorStatus()")
+
+	co_state, err := r.getOrCreateOperatorStatus()
```
[the remark this answers is not in the document]
Comment (jmencak) [reply pkg/controller/tuned/status.go]: Will change together with co_get.  The C-heritage...

**Comment by jmencak** [comment]
Hold the merge until the commits are squashed.
/hold

**Pull request merged**